autogpt/agent/agent.py

```diff
@@ -130,60 +172,45 @@ def start_interaction_loop(self):
-                    f"{self.ai_name}...",
+                    user_feedback_message,
                     flush=True,
                 )
                 while True:
                     console_input = clean_input(
                         Fore.MAGENTA + "Input:" + Style.RESET_ALL
                     )
-                    if console_input.lower().strip() == "y":
-                        user_input = "GENERATE NEXT COMMAND JSON"
-                        break
-                    elif console_input.lower().strip() == "s":
-                        logger.typewriter_log(
-                            "-=-=-=-=-=-=-= THOUGHTS, REASONING, PLAN AND CRITICISM WILL NOW BE VERIFIED BY AGENT -=-=-=-=-=-=-=",
-                            Fore.GREEN,
-                            "",
-                        )
-                        thoughts = assistant_reply_json.get("thoughts", {})
-                        self_feedback_resp = self.get_self_feedback(thoughts)
-                        logger.typewriter_log(
-                            f"SELF FEEDBACK: {self_feedback_resp}",
-                            Fore.YELLOW,
-                            "",
-                        )
-                        if self_feedback_resp[0].lower().strip() == "y":
-                            user_input = "GENERATE NEXT COMMAND JSON"
-                        else:
-                            user_input = self_feedback_resp
-                        break
-                    elif console_input.lower().strip() == "":
-                        print("Invalid input format.")
+
+                    new_command_name, self.next_action_count, user_input = \
+                        calculate_next_command_from_user_input(console_input)
+                    # If there was a parsing error, go back to the prompt
+                    # Otherwise if new command name is not None, update command name
+                    if new_command_name == "input error":
+                        print(user_input)
                         continue
-                    elif console_input.lower().startswith("y -"):
-                        try:
-                            self.next_action_count = abs(
-                                int(console_input.split(" ")[1])
-                            )
-                            user_input = "GENERATE NEXT COMMAND JSON"
-                        except ValueError:
-                            print(
-                                "Invalid input format. Please enter 'y -n' where n is"
-                                " the number of continuous tasks."
-                            )
-                            continue
-                        break
-                    elif console_input.lower() == "n":
-                        user_input = "EXIT"
-                        break
+                    elif new_command_name is not None:
+                        command_name = new_command_name
+                    break
```

Comment:
This break statement seems like it's going to kick you out of the loop on the first pass through. Have you run the application in its current state?